File: uri/2222/2222.c
/* Accepted */
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(){
    char **sets;
    int instances, setsQuantity, setSize, item, instructionsQuantity, instruction, set1, set2, result, i, j, k;

    scanf("%d", &instances);

    for( i = 0; i < instances; i++){
        scanf("%d", &setsQuantity);
        sets = malloc(setsQuantity * sizeof(char *));

        for ( j = 0; j < setsQuantity; j++){
            sets[j] = malloc(61 * sizeof(char));
            memset(sets[j],0,sizeof(char) * 61);
        }
        
        for ( j = 0; j < setsQuantity; j++){
            scanf("%d",&setSize);
            for ( k = 0; k < setSize; k++){
                scanf("%d", &item);
                sets[j][item] = 1;
            }
        }
        scanf("%d", &instructionsQuantity);
        for ( j = 0; j < instructionsQuantity; j++){
            scanf("%d %d %d",&instruction, &set1, &set2);
            result = 0;
            for ( k = 0; k < 61; k++)
            {
                if(instruction == 1 && sets[set1-1][k] && sets[set2-1][k]){
                    result++;
                }
                if(instruction == 2 && (sets[set1-1][k] || sets[set2-1][k])){
                    result++;
                }
            }
            printf("%d\n", result);
        }

        for ( j = 0; j < setsQuantity; j++){
            free(sets[j]);
        }   
    }

    return 0;
}

File: uri/2222/2222.js
/* time limit */
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

const instances = lines[0];

lines.shift();
lines.pop();

for (let i = 0; i < instances; i++) {
    const sets = [];
    const setsQuantity = lines[0];
    
    lines.shift();

    for (let j = 0; j < setsQuantity; j++) {
        const newSet = lines[j].split(' ');
        newSet.shift();
        sets.push(new Array(61).fill(0));
        newSet.forEach((v) => {
            sets[j][v] = 1;
        });
    }

    for (let j = 0; j < setsQuantity; j++) {
        lines.shift();
    }

    const instructionsQuantity = lines[0];

    lines.shift();
    
    for (let j = 0; j < instructionsQuantity; j++) {
        const [operation,set1, set2] = lines[j].split(' ');
        if(operation == 1){
            let result = 0;
            for (let k = 0; k < 61; k++) {
                if(sets[set1-1][k] && sets[set2-1][k]){
                    result++;
                }
            }
            console.log(result);
        }else{
            let result = 0;
            for (let k = 0; k < 61; k++) {
                if(sets[set1-1][k] || sets[set2-1][k]){
                    result++;
                }
            }
            console.log(result);
        }
    }

    for (let j = 0; j < instructionsQuantity; j++) {
        lines.shift();
    }
}